oioswift/common/middleware/autocontainerbase.py

```diff
@@ -60,35 +67,121 @@ def _convert_path(self, path):
             container, tail = split_path('/' + obj, 1, 2, True)
             obj = tail
 
-        if obj is None:
-            return account, container, None
-
-        container = quote_plus(self.con_builder(obj))
         return account, container, obj
 
-    def _convert_special_headers(self, account, env):
-        if 'HTTP_X_COPY_FROM' in env:
-            # HTTP_X_COPY_FROM_ACCOUNT will just pass through
-            if self.account_first:
-                src_path = "/fake_account" + env['HTTP_X_COPY_FROM']
-            else:
-                src_path = env['HTTP_X_COPY_FROM']
-            _, cont, obj = self._convert_path(src_path)
-            if not cont or not obj:
-                raise HTTPBadRequest(body="Malformed copy-source header")
-            env['HTTP_X_COPY_FROM'] = "/%s/%s" % (cont, obj)
-
-    def __call__(self, env, start_response):
-        if self.should_bypass(env):
-            return self.app(env, start_response)
+    def _convert_path(self, path):
+        account, container, obj = self._extract_path(path)
+        if obj is not None:
+            container = quote_plus(self.con_builder(obj))
+        return account, container, obj
 
+    def _alternatives(self, path):
+        account, container, obj = self._extract_path(path)
+        if obj is None:
+            yield account, container, obj
+            raise StopIteration
+        for alt_container in self.con_builder.alternatives(obj):
```

Comment:
Nothing checks that __ContainerBuilder__ has the function __alternatives()__ when __stop_at_first_match__ is set to True.